### ISSUE TITLE
FR: Partial page "turns" for continuous mode

### ISSUE THREAD

**Issue by jonnyl2** [koreader/koreader#10578]
**Does your feature request involve difficulty completing a task? Please describe.**

In continuous page view mode, it is often necessary or desirable to move the document by a few lines, e.g. to view an image/table/paragraph in its entirety on the screen. While this is of course doable via vertical panning, there are some drawbacks:

- The image rendering during panning is slow (especially on fixed-layout documents) and ugly 
- The panning gesture is sometimes not immediately recognized, or misinterpreted as text selection

**Describe the solution you'd like**

Allow partial page "turns" via gestures. E.g. +/- ½ page. (Or preferably, let users define the length in 0.1 page increments.)

**Describe alternatives you've considered**

A quick vertical swipe doesn't currently seem to have any assigned function (outside the menu areas, when activated). Alternatively (or additionally), allow a quick vertical swipe to move the document by the length of the swipe **upon release**. (This should be an optional feature and disabled by default as not everyone might need or want this.)

**Comment by hius07** [comment]
I think the current "Page overlap" feature can be used for this.
Can you try to increase the limit (line 61 in the file koreader/frontend/ui/elements/page_overlap.lua) and set the number of overlapped lines to (number of lines in the page less the desired page turn step, eg if the page has 25 lines and you want the 5 lines jump per turn, set it to 20).
If it works, we can implement this in the ui.

**Comment by jonnyl2** [comment]
>I think the current "Page overlap" feature can be used for this

This feature mainly seems to work for CRE documents? In most view settings for fixed-layout I wasn't able to enable Page overlap, and where I could (with horizontal overlap), it wasn't useful for this issue.

Personally, I am mostly interested in this FR for fixed-layout documents displayed in continuous mode zoomed to content width, as CRE docs usually display fine in Page view mode and tables/graphs are already rendered on their own separate screen pages.

**Comment by Frenzie** [comment]
> This feature mainly seems to work for CRE documents? 

No, not at all? Continuous mode there is just a thing that _exists_ for mysterious reasons.

**Comment by jonnyl2** [comment]
@Frenzie I was referring to "Page overlap".

**Comment by Frenzie** [comment]
@hius07 You seem to be referring to a different overlap (i.e., the one that draws an overlap which only applies in paged mode). I don't know why it's disabled in that context; try removing the restriction and see what happens.

But whatever the reasoning that means it works in crengine *by accident* because someone forgot to disable it as well then.

**Comment by hius07** [comment]
In creingine it works in continues mode only
https://github.com/koreader/koreader/blob/64f778d97d26a267a48c5a0cd123ba15e7e3bdd4/frontend/apps/reader/modules/readerview.lua#L1152-L1154

**Comment by jonnyl2** [comment]
To all (as a side note): there is the setting `DOVERLAPPIXELS` in the Advanced settings which could potentially be adapted into the UI for such partial page turns. However, it currently requires a restart to take effect, so I'm not sure if it could also be used while KOReader is running.

**Comment by offset-torque** [comment]
@jonnyl2 
> The image rendering during panning is slow (especially on fixed-layout documents) and ugly

To avoid this problem I set my **Scrolling** to **On-release scrolling**. Have you tried this ?
Actually in this mode it works exactly like you described: 

> a quick vertical swipe to move the document by the length of the swipe upon release

**Comment by jonnyl2** [comment]
@offset-torque Indeed, together with delay set to 0 ms it works exactly as I wished. Thank you for the advice! (How was I not aware of this setting?)

To all: I am happy with this resolution so I'm okay if this ticket is closed. But because there are ongoing discussions I will leave it open. There might still be a demand for partial page turns of a defined length via gestures by some users, but for me the quick swipes are perfect.

**Comment by Frenzie** [comment]
Tbh I thought that was already in gesture manager, but I guess only a number of pages is.

**Comment by n6id** [comment]
i am actually really confused about why im not seeing page overlap available for pdf continuous mode. "cannot be enabled in current view mode". it's definitely something i would want to use, with the greyed out lines, but im only able to activate it in paged mode. im on kindle paperwhite 6th gen with the 2024.11 release of koreader.
